presto-jdbc/src/test/java/io/prestosql/jdbc/BaseTestJdbcResultSet.java

```diff
@@ -591,4 +622,26 @@ public Statement getStatement()
             return statement;
         }
     }
+
+    private boolean serverSupportsVariablePrecisionTimestamp()
+    {
+        return getNumericPrestoServerVersion() >= 335;
+    }
+
+    private boolean serverSupportsVariablePrecisionTimestampWithTimeZone()
+    {
+        return getNumericPrestoServerVersion() >= 337;
+    }
+
+    private int getNumericPrestoServerVersion()
+    {
+        try {
+            return Integer.parseInt(prestoServerVersion.get());
+        }
+        catch (NumberFormatException e) {
+            log.info("Could not parse Presto version '%s' obtained from server; assuming server is recent");
+            // TODO it would be better to get version encoded in Manifest but it does not seem to be available for tests run via surefire/intellij
```

Comment:
The comment sounds as if author did not know why it is the case, and I am pretty sure you know that:
when running from IDE, classes are loaded from `*/target/classes` and there is no manifest there (it's jar's thing).

We could make it work by using resource filtering, but I do not see a need for this.

I'd advice improving comment wording here though

Reply:
Now I know :p

Reply:
Actually this comment does not make much sense at all, as we need a server version here. And what we could get from Manifest is client version. I think I kinda though that it is good fallback, not convinced it is any more.

Reply:
Let's split the method:

the subclasses testing against docker image will do parseInt and propagate failure 

the subclasses testing against local build can use MAX_INT or can read version from filtered resources
(to be totally sure this is server's version, you can add filtered resource in presto-main, where `TestingPrestoServer` is defined